api/rbac/v1alpha1/annotations.go

```diff
@@ -46,23 +49,50 @@ func OIDCClaimNameFromAnnotationKey(key string) (string, bool) {
 	return strings.TrimPrefix(key, AnnotationKeyOIDCClaimNamePrefix), true
 }
 
-// OIDCClaimsFromAnnotationValue parses the value of an
-// rbac.kargo.akuity.io/claims annotation and returns the corresponding list of
-// formatted and sorted claims.
-func OIDCClaimsFromAnnotationValue(value string) ([]string, error) {
+// OIDCClaimsFromAnnotationValue parses the values of both the
+// rbac.kargo.akuity.io/claims and rbac.kargo.akuity.io/claim.<name> annotations
+// and consolidates them into a single map where the value of each key is sorted and deduped.
+func OIDCClaimsFromAnnotationValues(annotations map[string]string) (map[string][]string, error) {
 	claims := make(map[string][]string)
```

Comment:
We can avoid an unnecessary allocation here.

```suggestion
	var claims map[string][]string
```

Reply:
You have to make the alloc here because making entries into `nil` map will cause a panic. 

You can repro this by running `TestOIDCClaimsFromAnnotationValues/old_style`

Reply:
Oh... I zeroed in on the `json.Unmarshal()` call below, for which you don't need to pre-allocate the map, _but I forgot that it's conditional_... if the new annotation isn't used, you reach the code starting on line +64 with a var that's still nil.

🤦‍♂️ Nothing to see here. Carry on.